### PR TITLE
Test case for sorting hidden files/folders

### LINKED ISSUE #3280: Hidden files on external storage with dotfiles crash the app
When enabling hidden files in settings and accessing an external storage (using SFTP) containing dotfiles, the Android app crashes.

### Actual behaviour

- The app crashes

### Expected behaviour

- The app should display a list of files

### Steps to reproduce

1. Have the following list of files and folders in the directory defined as root on an SFTP external storage

```
Folders:

.apache2
.cache
.config
.local
.logs
.nano
.nginx
.script-credentials
.ssh
.subversion
.znc
------------------------------
Files:

.bash_aliases
.bash_history
.bash_logout
.bashrc
.feral_aliases
.mysql_history
.profile
.selected_editor
.wget-hsts
```

2. Enable hidden files in the Nextcloud app settings.

3. Access the external storage

### Environment data

Android version: 8.1.2 or 7.0

Device model: Xiaomi Redmi Note 4 (mido), or also stock Samsung Galaxy A5 2016

Stock or customized system: Customized LineageOS 15.1, Stock "Samsung Experience".

Nextcloud app version: 3.3.2

Nextcloud server version: 14.0.3

#### Stacktrace

```
E/AndroidRuntime( 4641): FATAL EXCEPTION: main
E/AndroidRuntime( 4641): Process: com.nextcloud.client, PID: 4641
E/AndroidRuntime( 4641): java.lang.IllegalArgumentException: Comparison method violates its general contract!
E/AndroidRuntime( 4641): 	at java.util.TimSort.mergeHi(TimSort.java:899)
E/AndroidRuntime( 4641): 	at java.util.TimSort.mergeAt(TimSort.java:516)
E/AndroidRuntime( 4641): 	at java.util.TimSort.mergeCollapse(TimSort.java:441)
E/AndroidRuntime( 4641): 	at java.util.TimSort.sort(TimSort.java:245)
E/AndroidRuntime( 4641): 	at java.util.Arrays.sort(Arrays.java:1498)
E/AndroidRuntime( 4641): 	at java.util.ArrayList.sort(ArrayList.java:1470)
E/AndroidRuntime( 4641): 	at java.util.Collections.sort(Collections.java:201)
E/AndroidRuntime( 4641): 	at com.owncloud.android.utils.FileSortOrderByName.sortCloudFiles(FileSortOrderByName.java:52)
E/AndroidRuntime( 4641): 	at com.owncloud.android.ui.adapter.OCFileListAdapter.swapDirectory(OCFileListAdapter.java:513)
E/AndroidRuntime( 4641): 	at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1080)
E/AndroidRuntime( 4641): 	at com.owncloud.android.ui.fragment.OCFileListFragment.onItemClicked(OCFileListFragment.java:809)
E/AndroidRuntime( 4641): 	at com.owncloud.android.ui.adapter.OCFileListAdapter.lambda$onBindViewHolder$0$OCFileListAdapter(OCFileListAdapter.java:276)
E/AndroidRuntime( 4641): 	at com.owncloud.android.ui.adapter.OCFileListAdapter$$Lambda$0.onClick(Unknown Source:4)
E/AndroidRuntime( 4641): 	at android.view.View.performClick(View.java:6294)
E/AndroidRuntime( 4641): 	at android.view.View$PerformClick.run(View.java:24770)
E/AndroidRuntime( 4641): 	at android.os.Handler.handleCallback(Handler.java:790)
E/AndroidRuntime( 4641): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 4641): 	at android.os.Looper.loop(Looper.java:164)
E/AndroidRuntime( 4641): 	at android.app.ActivityThread.main(ActivityThread.java:6494)
E/AndroidRuntime( 4641): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 4641): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
E/AndroidRuntime( 4641): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

```

#### Web server error log

```
Error log stays empty when reproducing the issue (and five minutes afterwards).

```

#### Nextcloud log (data/nextcloud.log)

```
Error log stays empty when reproducing the issue (and five minutes afterwards).

```

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Resolves #3280 - adding a tes case for dot files/folders 

Fix has already happened for 3.4.0 in an older PR while this adds another test case to check this via CI.

### CODE REVIEW

File: src/test/java/com/owncloud/android/utils/TestSorting.java [hunk 1]
```diff
@@ -372,6 +372,34 @@ public void testSortCloudFilesWithDots() {
         assertTrue(sortAndTest(array));
     }
 
+    @Test
+    public void testSortCloudFilesWithDotFilesAndFolders() {
+        List<OCFile> unsortedArray = new ArrayList<>();
```
Comment (tobiasKaminsky) [inline src/test/java/com/owncloud/android/utils/TestSorting.java]: This is a sorted array :-D

Comment (AndyScherzinger) [reply src/test/java/com/owncloud/android/utils/TestSorting.java]: Renamed and rebased :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I renamed the variable according to your comment and did a rebase, ready for final review I guess 🎉

**Comment by tobiasKaminsky** [comment]
:+1:

**Pull request merged**